### PR TITLE
Add coverage in Makefile and fix compile warning.

### PR DESCRIPTION
**Pull request by zhangh43**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,6 +22,8 @@ OBJS = $(foreach FILE,$(FILES),$(subst .c,.o,$(FILE)))
 PGXS := $(shell pg_config --pgxs)
 include $(PGXS)
 
+CLIENT_CFLAG = -Werror -Wextra -Wall
```
Comment (paul-guo-) [inline Makefile]: Why we do this change?

File: Makefile [hunk 2]
```diff
@@ -58,13 +60,14 @@ endif
 PLCONTAINERDIR = $(DESTDIR)$(datadir)/plcontainer
 
 override CFLAGS += -Werror -Wextra -Wall
+override CFLAGS += $(COVERAGE_FLAG)
+override SHLIB_LINK += $(COVERAGE_LINK_FLAG)
```
Comment (HaozhouWang) [inline Makefile]: Will we need to re-build GPDB for coverage flag?

Comment (zhangh43) [reply Makefile]: we use gpdb extension(PGXS := $(shell pg_config --pgxs) include $(PGXS)) to build plcontainer.
And to enable code coverage, we need add compile and link flag.

File: Makefile (hunk 1 above)
Comment (zhangh43) [reply Makefile]: Client should not inherit the flag of plcontainer.so.
In future we may enable code coverage on client side

File: Makefile (hunk 2 above)
Comment (HaozhouWang) [reply Makefile]: Sure. Thanks

**Pull request merged**